### ISSUE TITLE
SSL failed with valid certificate

### ISSUE THREAD

**Issue by jean-io** [owncloud/client#2430]
Hi all,

I can not connect to my server from my mac. But it works fine on my Android. 

I get this error:
![2014-10-29 09 21 40 pm](https://cloud.githubusercontent.com/assets/4070933/4833964/3e46fc80-5fa9-11e4-930c-7ae838dd7bf2.png)
- client version: 1.6.4
- server version: 7.0.2.1
- [owncloud.jmercadier.fr](https://owncloud.jmercadier.fr) use a valid SSL certificate from CloudFlare
- I am really tired of webdav, please help

`status.php` return this:

```
{"installed":"true","version":"7.0.2.1","versionstring":"7.0.2","edition":""}
```

Here is the log file:

```
10-29 21:18:45:829 account.cpp:135 Migrate: checking old config  "/Users/Jean/Library/Application Support/ownCloud/owncloud.cfg" 
10-29 21:18:45:838 owncloudwizard.cpp:160 Current Wizard page changed to  0 
10-29 21:18:52:315 cookiejar.cpp:119 "/Users/Jean/Library/Application Support/ownCloud//cookies.db" 
10-29 21:18:52:315 networkjobs.cpp:75 void Mirall::AbstractNetworkJob::setTimeout(qint64) 10000 
10-29 21:18:52:316 networkjobs.cpp:201 !!! Mirall::CheckServerJob created for  QUrl( "http://owncloud.jmercadier.fr/" )  querying "status.php" 
10-29 21:18:52:960 networkjobs.cpp:146 void Mirall::AbstractNetworkJob::slotFinished() 6 "SSL handshake failed" 
10-29 21:18:52:961 networkjobs.cpp:424 status.php from server is not valid JSON! 
10-29 21:18:52:961 networkjobs.cpp:427 status.php returns:  QMap()   6  Reply:  QNetworkReplyHttpImpl(0x7fb8e9548180) 
10-29 21:18:52:961 networkjobs.cpp:433 No proper answer on   QUrl( "https://owncloud.jmercadier.fr/status.php" )  
10-29 21:19:16:314 owncloudwizard.cpp:160 Current Wizard page changed to  -1 
10-29 21:19:16:317 owncloudsetupwizard.cpp:401 Rejected the new config, use the old! 
10-29 21:19:16:319 cookiejar.cpp:119 "/Users/Jean/Library/Application Support/ownCloud//cookies.db" 
```

**Comment by garrettr** [comment]
Can this issue be re-opened? It was closed due to a "Can't reproduce" in https://github.com/owncloud/client/issues/2430#issuecomment-61368967 but I can confirm that this is still an issue - the desktop client does not sync if the OwnCloud server is using Cloudflare. It is probably confused by Cloudflare's unusual (but valid) certificates, and/or its use of SNI.

I am happy to help debug in any way that might be useful (can provide logs from a broken client as well as from the OwnCloud server).

**Comment by Ricain** [comment]
@garrettr what error message do you have? which version of owncloud are you using on which OS?

**Comment by garrettr** [comment]
@Ricain Sorry for the delay in responding! We are currently use the same workaround as https://github.com/owncloud/client/issues/2430#issuecomment-64486966. I am disabling it now and attempting to connect so I can get a copy of the error message for you.

**Comment by garrettr** [comment]
@Ricain 

Platform: Mac 10.10.3
OwnCloud Version: 1.8.1 (Build 2335)

I turned Cloudflare back on for my OwnCloud subdomain and tried to reproduce the error. However, it appears that the problem has been fixed. I am able to successfully connect, and have confirmed that the client is using the Cloudflare certificate.

![owncloud_cloudflare](https://cloud.githubusercontent.com/assets/407302/8042889/26fdbfb4-0dd7-11e5-9b73-d60fec4cdb29.png)


I am not sure what changed that caused it to start working, but I'm glad it's working anyway :relieved:
